javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,360 @@
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => arr.reduce((acc, value) => acc + value, 0);
+ 
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addPurchases that, given an array of objects as input, uses reduce to find the total amount purchased. Each object contains the keys `item` and `purchasePrice` like the example.
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  return arr.reduce((acc, purchasePrice) => {
+    return acc + purchasePrice.purchasePrice;
+  }, 0);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  return arr.reduce((acc, items) => {
+    acc = items++;
+    return acc;
+  }, 0);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named returnNames that, given the Star Wars data, below, uses reduce to return an array containing the names of the characters.
+------------------------------------------------------------------------------------------------ */
+
+let starWarsData = [{
+  name: 'Luke Skywalker',
+  height: '172',
+  mass: '77',
+  hair_color: 'blond',
+  skin_color: 'fair',
+  eye_color: 'blue',
+  birth_year: '19BBY',
+  gender: 'male',
+},
+{
+  name: 'C-3PO',
+  height: '167',
+  mass: '75',
+  hair_color: 'n/a',
+  skin_color: 'gold',
+  eye_color: 'yellow',
+  birth_year: '112BBY',
+  gender: 'n/a'},
+{
+  name: 'R2-D2',
+  height: '96',
+  mass: '32',
+  hair_color: 'n/a',
+  skin_color: 'white, blue',
+  eye_color: 'red',
+  birth_year: '33BBY',
+  gender: 'n/a'
+},
+{
+  name: 'Darth Vader',
+  height: '202',
+  mass: '136',
+  hair_color: 'none',
+  skin_color: 'white',
+  eye_color: 'yellow',
+  birth_year: '41.9BBY',
+  gender: 'male'
+},
+{
+  name: 'Leia Organa',
+  height: '150',
+  mass: '49',
+  hair_color: 'brown',
+  skin_color: 'light',
+  eye_color: 'brown',
+  birth_year: '19BBY',
+  gender: 'female'
+}];
+
+const returnNames = (arr) => {
+  let newArr = [];
+  return arr.reduce((acc, names) => {
+    newArr.push(acc = names.name);
+    return newArr;
+  }, []);
+};
+
```

Comment:
Really great work on all the challenges!

Another way we can do this,  since we are returning an arr in reduce(), is to just push `names.name` directly into the acc and then just return the acc on line 120 👍 You could in this case remove line 117.

```
return arr.reduce((acc, names) => {
    acc.push(names.name);
    return acc;
  }, []);
````